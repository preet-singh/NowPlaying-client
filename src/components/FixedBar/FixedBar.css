@media screen and (min-width: 320px){
  .fixed-bar {
    border-top: 1px solid grey;
    padding: 10px;
    width: 100%;
    height: 260px;
    margin: auto;
    position: fixed;
    bottom: 0px;
    right: 0px;
    background-color: rgb(58, 175, 58);
    z-index: 2;
  }
  
  .fixed-bar-header {
    margin-top: -20px;
    margin-bottom: 10px;
    border: none;
    padding: 10px;
    position: relative;
    padding-left: -140px;
  }
  
  .fixed-bar-header h3 {
    position: initial;
  }

  .fixed-bar-header button {
    border: none;
  }
}

@media screen and (min-width: 768px){
  .fixed-bar {
    min-width: 445px;
    height: 300px;
    max-width: 30%;
    position: fixed;
    bottom: 0px;
    right: 450px;
    margin-right: 445px;
  }

  .fixed-bar-header {
    margin-top: 0px;
    margin-left: 0px;
    padding: 0px;
    position: relative;
    text-align: center;
  }

  .fixed-bar-header h3 {
    position: initial;
  }

  .fixed-bar-header button {
    position: absolute !important;
    z-index: 3;
    top: -25px !important;
    right: 45px !important;
  }

  .fixed-bar-header button:hover {
    cursor: pointer;
  }

}