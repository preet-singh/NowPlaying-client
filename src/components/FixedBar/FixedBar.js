import React from 'react';
import './FixedBar.css'
import AddCommentBox from '../../components/AddCommentBox/AddCommentBox';
import PlayButton from '../../components/PlayButton/PlayButton'
import UserContext from '../../utils/context';

class FixedBar extends React.Component {
  static contextType = UserContext;

  componentDidMount() {
    this.myInterval = setInterval(() => this.context.updateMediaTimer(), 1000)
  }

  componentWillUnmount() {
    clearInterval(this.myInterval);
  }

  convertTimeString = timeValue => {
    if(timeValue < 10) {
      return `0${timeValue}`;
    }
    else {
      return `${timeValue}`;
    }
  }

  convertSeconds = seconds => {
    let secondsValue = seconds % 60;
    let hoursValue = seconds > 3600 ? (seconds - (seconds % 3600)) / 3600 : 0;
    let minutesValue = seconds < 3600 ? (seconds - secondsValue) / 60 : (seconds - secondsValue) % 3600 / 60
    let timeArray = [hoursValue, minutesValue, secondsValue]
    let timeStringArray = timeArray.map(time => this.convertTimeString(time))

    let timeString = timeStringArray.join(':')
    return timeString;
  }

  render() {
    if (this.context.playing) {
    return (
      <div className='fixed-bar'>
        <div className='fixed-bar-header'>
        <PlayButton />
        <h3 id='media-timer'>{this.convertSeconds(this.context.mediaTimer)}</h3>
        </div>
        <AddCommentBox category={this.props.category} mediaId={this.props.mediaId}/>
      </div>
    )
  }
  else {
    return null;
}
}
}

export default FixedBar;