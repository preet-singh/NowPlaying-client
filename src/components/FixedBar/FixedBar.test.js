import React from 'react';
import ReactDOM from 'react-dom';
import FixedBar from './FixedBar';
import { MemoryRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'

describe.only('Fixed bar testing', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <MemoryRouter>
            <FixedBar />
        </MemoryRouter>, 
    div);
    ReactDOM.unmountComponentAtNode(div);
  });
  
  it('renders the play button within the fixed bar', () => {
    const wrapper = shallow(<FixedBar />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
  
});