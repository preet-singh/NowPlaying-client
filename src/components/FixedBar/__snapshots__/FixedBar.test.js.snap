// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fixed bar testing renders the play button within the fixed bar 1`] = `
<div
  className="fixed-bar"
>
  <div
    className="fixed-bar-header"
  >
    <h2>
      <Link
        to="/undefined/undefined"
      />
    </h2>
    <PlayButton />
    <h3
      id="media-timer"
    >
      00:NaN:NaN
    </h3>
  </div>
  <AddCommentBox />
   }
</div>
`;
